apps/acqdat_core/test/entity-management/model/sensor_test.exs

```
@@ -0,0 +1,126 @@
+
+      assert {:ok, _sensor} = SensorModel.create(params)
+    end
+
+    test "fails if org_id is not present", context do
+      %{project: project} = context
+
+      params = %{
+        name: "Demo sensor",
+        project_id: project.id
+      }
+
+      assert {:error, changeset} = SensorModel.create(params)
+      assert %{org_id: ["can't be blank"]} == errors_on(changeset)
+    end
+
+    test "fails if name is not present", context do
+      %{org: org, project: project} = context
+
+      params = %{
+        org_id: org.id,
+        project_id: project.id
+      }
+
+      assert {:error, changeset} = SensorModel.create(params)
+      assert %{name: ["can't be blank"]} == errors_on(changeset)
+    end
+
+    test "fails if project is not present", context do
+      %{org: org} = context
+
+      params = %{
+        org_id: org.id,
+        name: "Demo sensor Test"
+      }
+
+      assert {:error, changeset} = SensorModel.create(params)
+      assert %{project_id: ["can't be blank"]} == errors_on(changeset)
+    end
+  end
+
+  describe "update/2" do
+    setup do
+      project = insert(:project)
+      sensor = insert(:sensor)
+
+      [sensor: sensor, project: project]
+    end
+
+    test "updates the sensor's name", context do
+      %{sensor: sensor} = context
+
+      params = %{
+        name: "updated sensor name"
+      }
+
+      assert {:ok, sensor} = SensorModel.update(sensor, params)
+      assert sensor.name == "updated sensor name"
+    end
+
+    test "updates the sensor's parent as project", context do
+      %{sensor: sensor, project: project} = context
+
+      params = %{
+        parent_id: project.id,
+        parent_type: "Project"
+      }
+
+      assert {:ok, sensor} = SensorModel.update(sensor, params)
+      assert sensor.parent_id == project.id
+      assert sensor.parent_type == "Project"
+    end
+  end
+
+  describe "delete/2" do
+    setup do
+      sensor = insert(:sensor)
+
+      [sensor: sensor]
+    end
+
```

Add a test if a sensor is deleted with data then it should raise an error.